### PR TITLE
LCH-5852: Get list of all sites from ACSF API instead of first 10.

### PR DESCRIPTION
**Pull request by amangrover90**

### CODE REVIEW

File: src/Client/AcsfClient.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function ping(): bool {
    *   All the existing site the user has permission to view.
    */
   public function listSites(): array {
-    $response = $this->getJsonResponseBody($this->get('sites'), 'No sites found.');
+    $response = $this->getJsonResponseBody($this->get('sites?limit=1000'), 'No sites found.');
```
Comment (ankitsingh0188) [inline src/Client/AcsfClient.php:66]: Let's include $limit as function param with default value and use it as a query param.

**Review by ankitsingh0188 — APPROVED** [review]
Looks good to me!!

File: src/Client/AcsfClient.php [hunk 2]
```diff
@@ -59,11 +59,14 @@ public function ping(): bool {
...
+   * @param int $limit
+   *   Query limit.
+   *
    * @return array
    *   All the existing site the user has permission to view.
    */
-  public function listSites(): array {
-    $response = $this->getJsonResponseBody($this->get('sites'), 'No sites found.');
+  public function listSites(int $limit = 1000): array {
+    $response = $this->getJsonResponseBody($this->get("sites?limit={$limit}"), 'No sites found.');
```
Comment (saxumVermes) [inline src/Client/AcsfClient.php:69]: We should be using the API properly here:
```suggestion
    $response = $this->getJsonResponseBody($this->get('sites', [
      RequestOptions::QUERY => ['limit' => $limit]
    ]), 'No sites found.');
```

File: .github/workflows/acsf-console-actions.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ jobs:
       - name: Install composer dependencies
         run: |
           composer require --dev php-coveralls/php-coveralls
+          composer require --dev drupal/coder
```
Comment (saxumVermes) [inline .github/workflows/acsf-console-actions.yml:26]: Could you please add this to composer.json as well to the dev deps? And by that, could you please confirm that without requiring the package here, a simple `composer install` will not do the work, I.e. do you still need this line even after adding the package to dev deps?

**Review by saxumVermes — CHANGES_REQUESTED** [review]

**Review by saxumVermes — COMMENTED** [review]
I didn't test it functionally, @ankitsingh0188  have you by any chance?

File: .github/workflows/acsf-console-actions.yml (hunk 3 above)
Comment (amangrover90) [reply .github/workflows/acsf-console-actions.yml:26]: We already have it in helpers' composer.json in require-dev section, do you think we need it in all the libraries separately?

Comment (saxumVermes) [reply .github/workflows/acsf-console-actions.yml:26]: Yes, if I'm not mistaken, composer will not install the dev deps of dependencies by default.

Comment (saxumVermes) [reply .github/workflows/acsf-console-actions.yml:26]: And if you include drupal/coder in composer.json this won't be needed.

Comment (amangrover90) [reply .github/workflows/acsf-console-actions.yml:26]: Yes you are right, dev dependencies need to be present for each package separately. I'll test the scenario with composer.json.

Comment (amangrover90) [reply .github/workflows/acsf-console-actions.yml]: @saxumVermes drupal/coder gets installed if added to require-dev in composer.json but it starts to throw a different error on Github Actions:
`Run ./vendor/bin/phpcs --config-set installed_paths vendor/drupal/coder/coder_sniffer
Using config file: /home/runner/work/acsf-contenthub-console/acsf-contenthub-console/vendor/squizlabs/php_codesniffer/CodeSniffer.conf

Config value "installed_paths" updated successfully; old value was "../../drupal/coder/coder_sniffer,../../sirbrillig/phpcs-variable-analysis,../../slevomat/coding-standard"
ERROR: Referenced sniff "SlevomatCodingStandard.ControlStructures.RequireNullCoalesceOperator" does not exist

Run "phpcs --help" for usage information

Error: Process completed with exit code 3.`

Same error appears for helpers library as well. If added to workflow yml it works fine. Do you think we can go that way where we remove this from composer.json from helpers library as well and just add it dynamically on GA.

Comment (saxumVermes) [reply .github/workflows/acsf-console-actions.yml]: @amangrover90  I think it should be good now.

**Review by saxumVermes — APPROVED** [review]

**Comment by ankitsingh0188** [comment]
> I didn't test it functionally, @ankitsingh0188 have you by any chance?

Yeah

**Pull request merged**